tests/test_validate_artifacts.py

```diff
@@ -0,0 +1,35 @@
+import json
+import os.path
+import unittest
+from assertpy import assert_that, fail
+import jsonschema
+
+MODEL_PATH = 'crdch_model'
+
+
+class ValidateArtifacts(unittest.TestCase):
+    """
+    This is a series of tests for validating that artifacts have been generated correctly.
+    """
+
+    def test_validate_json_schema(self):
+        json_schema_file_path = os.path.join(MODEL_PATH, 'json_schema', 'crdch_model.schema.json')
+        assert_that(json_schema_file_path).exists().is_file()
+
+        with open(json_schema_file_path) as f:
+            try:
+                json_schema = json.load(f)
+            except json.JSONDecodeError as e:
+                fail(f'JSON Schema {json_schema_file_path} is not a valid JSON document: {e}')
+
+            assert_that(json_schema).is_not_empty()
+
+            # Idea from https://github.com/Julian/jsonschema/issues/348#issuecomment-647418176
+            class SaferDraft202012Validator(jsonschema.Draft202012Validator):
```

Comment:
@gaurav: is there any specific reason why we want to use this `SaferDraft202012Validator` class rather than just `Draft202012Validator` or `Draft7Validator`?

Reply:
Why not `Draft7Validator`: because it uses `definitions` rather than `$defs`, which is what we're trying to move to.
Why not `Draft202012Validator`: because by default `Draft202012Validator` sets `additionalProperties` to true, which makes the validator _very_ permissible -- any property that doesn't validate is assumed to be an additional property, and validation is successful :). In retrospect, we should probably call this `LessPermissiveDraft202012Validator` or something.